api/nodes/serializers.py

```diff
@@ -1721,3 +1726,116 @@ def enable_or_disable_addon(self, obj, should_enable, addon_name, auth):
         if type(addon) == bool:
             addon = None
         return addon
+
+
+class NodeGroupsSerializer(JSONAPISerializer):
+    filterable_fields = frozenset([
+        'name',
+        'permission',
+        'date_created',
+    ])
+
+    writeable_method_fields = frozenset([
+        'permission',
+    ])
+
+    id = CompoundIDField(source='_id', read_only=True)
+    type = TypeField()
+    permission = ser.SerializerMethodField()
+    name = ser.CharField(read_only=True)
+    date_created = VersionedDateTimeField(source='created', read_only=True)
+    date_modified = VersionedDateTimeField(source='modified', read_only=True)
+
+    osf_groups = RelationshipField(
+        related_view='osf_groups:group-detail',
+        related_view_kwargs={'group_id': '<_id>'},
+        required=False,
+    )
+
+    links = LinksField({
+        'self': 'get_absolute_url',
+    })
+
+    def get_absolute_url(self, obj):
+        node = self.context['view'].get_node()
+        return absolute_reverse(
+            'nodes:node-group-detail', kwargs={
+                'group_id': obj._id,
+                'node_id': node._id,
+                'version': self.context['request'].parser_context['kwargs']['version'],
+            },
+        )
+
+    def get_permission(self, obj):
+        node = self.context['view'].get_node()
+        permissions = self.context['view'].get_node_group_perms(obj, node)
+        return osf_permissions.reduce_permissions(permissions)
+
+    class Meta:
+        type_ = 'node_groups'
+
+
+class NodeGroupsCreateSerializer(NodeGroupsSerializer):
+    """
+    Overrides NodeGroupSerializer so osf_groups relationship is properly parsed
+    (JSONAPIParser will flatten osf_groups relationship into {'_id': 'group_id'},
+    so _id field needs to be writeable so it's not dropped from validated_data)
+
+    """
+    id = IDField(source='_id', required=False, allow_null=True)
+
+    osf_groups = RelationshipField(
+        related_view='osf_groups:group-detail',
+        related_view_kwargs={'group_id': '<_id>'},
+        required=False,
+    )
+
+    def load_osf_group(self, _id):
+        if not _id:
+            raise exceptions.ValidationError(detail='OSFGroup relationship must be specified.')
```

Comment:
Same here: I don't think we should expos the "OSFGroup" language to clients. "Group" is fine here.

Reply:
Yep, thanks, I'll change all the instances where we're saying OSFGroup externally to Group language, in both endpoints and errors.